version: '3.8'

services: #Servicios que corremos en nuestra red de docker.
  app:
    build:
      context: .  #AQUI VIENEN LOS PROBLEMAS :) ./docker/python
      dockerfile: Dockerfile
    ports:
      - '5000:5000'
    environment:
      - FLASK_ENV=development
    #volumes:
    #  - .:/app
    depends_on:
      - db

  db:
    build:
      context: ./docker/db
      dockerfile: Dockerfile
    environment:
      POSTGRES_DB: microblogging
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password
    volumes:
      - postgres_data:/var/lib/postgresql/data

  pgadmin:
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: "admin@admin.com"
      PGADMIN_DEFAULT_PASSWORD: "root"
    ports:
      - "8080:80"
    depends_on:
      - db

volumes:
  postgres_data: